### ISSUE TITLE
NullPointerException in XHRMultipartTransport.java

### ISSUE THREAD

**Issue by itaylor-cf** [tadglines/Socket.IO-Java#6]
Hi guys,

I've been playing around with this library since yesterday and I'm getting the following exception:

```
2011-02-14 20:48:50.789:WARN::/monitor/socket.io/xhr-multipart/
java.lang.NullPointerException at com.glines.socketio.server.transport.XHRMultipartTransport$XHRMultipartSessionHelper.writeData(XHRMultipartTransport.java:66)
    at com.glines.socketio.server.transport.XHRMultipartTransport$XHRMultipartSessionHelper.customConnect(XHRMultipartTransport.java:83)
    at com.glines.socketio.server.transport.XHRTransport$XHRSessionHelper.connect(XHRTransport.java:330)
    at com.glines.socketio.server.transport.XHRTransport.connect(XHRTransport.java:396)
    at com.glines.socketio.server.transport.AbstractHttpTransport.handle(AbstractHttpTransport.java:99)
    at com.glines.socketio.server.SocketIOServlet.serve(SocketIOServlet.java:117)
    at com.glines.socketio.server.SocketIOServlet.doGet(SocketIOServlet.java:85)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:734)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:546)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:483)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:479)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:230)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:956)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:411)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:188)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:891)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:247)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:151)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:114)
    at org.eclipse.jetty.server.Server.handle(Server.java:353)
    at org.eclipse.jetty.server.HttpConnection.handleRequest(HttpConnection.java:598)
    at org.eclipse.jetty.server.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:1059)
    at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:590)
    at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:212)
    at org.eclipse.jetty.server.HttpConnection.handle(HttpConnection.java:427)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:510)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint.access$000(SelectChannelEndPoint.java:34)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:40)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:450)
    at java.lang.Thread.run(Thread.java:680)
```

I'm working around it by removing xhr-multipart from my list of transports.
I think the problem is either that the idleCheck field is final and is being set as null on the first request during the initialization process, or that the 
    public static IdleCheck newTimeoutPreventor() {...
method in ConnectionTimeoutPreventer is returning null in a way that is unexpected.  

If you'd like me to look into it further, I'd be glad to get a debugger on it and create a patch...

**Comment by yurize** [comment]
Actually, I think I also see similar exceptions in the Wave in a Box logs (which uses Socket.IO for handling Web Sockets).
Any progress on this?

**Comment by itaylor-cf** [comment]
I ended up not using Socket-io-java... despite being very good idea, the project seems to lack momentum. I introduced node.js to my project and used real socket.io instead for the parts that required websocket.  Sorry if that's not very helpful.   

On this particular issue, I believe removing final from idleCheck fixed it for me, but I gave up on using the project a few days after that.

If you're considering replacing Socket-io-java with another Java Solution, I'd recommend taking a good look at Atmosphere... https://github.com/Atmosphere/atmosphere

Good luck!
